File: kamon-examples/kamon-annotation-example/java/src/main/java/kamon/annotation/KamonController.java
package kamon.annotation;

import org.springframework.boot.autoconfigure.*;
import org.springframework.stereotype.*;
import org.springframework.web.bind.annotation.*;

@Controller
@EnableAutoConfiguration
@RequestMapping("/kamon")
@EnableKamon
public class KamonController {

    @RequestMapping("/counter")
    @ResponseBody
    @Count(name = "awesomeCounter")
    public String counter() {  return "count!!!"; }
}

File: kamon-examples/kamon-annotation-example/scala/src/main/scala/kamon/annotation/KamonController.scala
package kamon.annotation

import org.springframework.boot.autoconfigure._
import org.springframework.stereotype._
import org.springframework.web.bind.annotation._

@Controller
@EnableAutoConfiguration
@RequestMapping(Array("/kamon"))
@EnableKamon
class KamonController {

    @RequestMapping(Array("/counter"))
    @ResponseBody
    @Count(name = "awesomeCounter")
    def counter(): String =  "count!!!"
}